CALIDAD DE AGUA DEL RÍO DE LA PLATA - 2021
     
Este conjunto de datos incluye 166 mediciones en diferentes sitios del Río de La Plata durante 2021, 
y en general, uno en cada estación del año. 
Descripción de las variables
1. sitios: Localización específica donde se realizó el muestreo del agua.
2. codigo: Identificador único para cada muestra o estación de muestreo.
3. fecha: Fecha en la que se tomó la muestra de agua.
4. año: Año en que se realizó el muestreo.
5. campaña: Nombre o número de la campaña de monitoreo en la que se realizó el muestreo.
6. tem agua: Temperatura del agua en grados Celsius.
7. tem aire: Temperatura del aire en grados Celsius.
8. od: Oxígeno disuelto, medido en miligramos por litro (mg/L), esencial para la vida acuática.
9. ph: Medida de la acidez o alcalinidad del agua, en una escala de 0 a 14.
10. olores: Presencia de olores en el agua, que puede indicar contaminación.
11. color: Color del agua, que puede ser un indicador de la calidad del agua.
12. espumas: Presencia de espumas en la superficie del agua, que puede ser un signo de contaminación.
13. mat susp: Materia suspendida, que se refiere a partículas sólidas que flotan en el agua.
14. colif fecales ufc 100ml: Unidades formadoras de colonias de coliformes fecales en 100 ml de agua, un indicador de contaminación fecal.
15. escher coli ufc 100ml: Unidades formadoras de colonias de Escherichia coli en 100 ml de agua, otro indicador de contaminación fecal.
16. enteroc ufc 100ml: Unidades formadoras de colonias de enterococos en 100 ml de agua, que también indican contaminación fecal.
17. nitrato mg l: Concentración de nitratos en miligramos por litro (mg/L), que puede indicar contaminación por fertilizantes.
18. nh4 mg l: Concentración de amonio en miligramos por litro (mg/L), que puede ser un indicador de contaminación orgánica.
19. p total l mg l: Fósforo total en miligramos por litro (mg/L), que incluye todas las formas de fósforo en el agua.
20. fosf ortofos mg l: Concentración de ortofosfatos en miligramos por litro (mg/L), que es un nutriente importante.
21. dbo mg l: Demanda biológica de oxígeno en miligramos por litro (mg/L), que mide la cantidad de oxígeno requerido por microorganismos para descomponer materia orgánica.
22. dqo mg l: Demanda química de oxígeno en miligramos por litro (mg/L), que mide la cantidad total de oxígeno requerido
     para oxidar materia orgánica e inorgánica.
23. turbiedad ntu: Turbidez del agua medida en unidades NTU (Nephelometric Turbidity Units), que indica la claridad del agua.
24. hidr deriv petr ug l: Hidrocarburos derivados del petróleo en microgramos por litro (µg/L), que indican contaminación por productos petroleros.
25. cr total mg l: Concentración total de cromo en miligramos por litro (mg/L), un metal pesado que puede ser tóxico.
26. cd total mg l: Concentración total de cadmio en miligramos por litro (mg/L), otro metal pesado que es tóxico en altas concentraciones.
27. clorofila a ug l: Concentración de clorofila a en microgramos por litro (µg/L), que indica la cantidad de fitoplancton en el agua.
28. microcistina ug l: Concentración de microcistinas en microgramos por litro (µg/L), que son toxinas producidas por ciertas algas.
29. ica: Índice de calidad del agua, que puede ser un valor calculado para evaluar la calidad general del agua.
30. calidad de agua: Clasificación general de la calidad del agua basada en los parámetros medidos.

Empezamos armando el dataframe con el que vamos a trabajar:

In [78]:
import pandas as pd

# Leer el archivo csv original especificando el delimitador como ';'
df_original = pd.read_csv("Calidad_de_agua_2021.csv", sep=';')

# Creamos una copia del DataFrame
df = df_original.copy()

In [79]:
#Mostramos el df para ver si cargo bien
#df

In [80]:
#Esto es para que nos muestre todas las columnas y las filas
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

In [81]:
# Le hacemos un info para indentificar los tipos

#df.info(verbose=True)

Empezamos a ver los tipos y modificarlos para que sean datos mas manejables 

In [82]:
#df['espumas'].value_counts(dropna=False)

In [83]:
# Aca identificamos las que estan escritas dif (para saber sus diferencias tambien)
#for idx in df["espumas"].value_counts().items():
#    print(idx)

In [84]:
# Remplazo los valores q estan escritos distintos para q esten iguales
df['espumas'] = df['espumas'].replace("Ausencia ", 'Ausencia')
df['espumas'] = df['espumas'].replace("no se midió", pd.NA)


In [85]:
# Aca miramos la cantidad de NaN y miramos cuales los los tipos que contiene
#print(df["espumas"].isna().sum())
#print(df["espumas"].unique())

In [86]:
#Los pasamos a Booleano para mejor manupilacion  y los nulos los ignoramos para que sigan estando
df['espumas'] = df['espumas'].map({'Presencia': True, 'Ausencia': False}, na_action= 'ignore')


In [87]:
#Verificamos que efectivamente no se borraron los NaN
#df["espumas"].isna().sum()

In [88]:
#Empezamos a curar otra columna, este caso "fecha"
#df['fecha'].value_counts(dropna=False)

In [89]:
#pasamos la fecha a formato date
df['fecha'] = pd.to_datetime(df['fecha'], format='%d/%m/%Y')

In [90]:
# Miramos el contenido de campaña 
#df['campaña'].value_counts()

In [91]:
# Miramos el contenido de la temperatura del agua 
#df['tem_agua'].value_counts(dropna=False)

In [92]:
# Reemplazas a los "no se midio" por NaN y dsp lo pasa a tipo numerico, en este caso uso el ignore porque quiero 
# asegurarme que unicamente los "no se midio" se esten cambiando
df['tem_agua'] = df['tem_agua'].replace("no se midió", pd.NA)
df['tem_agua'] = pd.to_numeric(df['tem_agua'] )# , errors='ignore')



In [93]:
# Hacemos lo mismo con los datos del aire 
df['tem_aire'] = df['tem_aire'].replace("no se midió", pd.NA)
df['tem_aire'] = pd.to_numeric(df['tem_aire']) #, errors='ignore')

In [94]:
# Miramos que datos tienen 
#df['ica'].value_counts(dropna=False)

In [95]:
#  Cambiamos los no se midio por NaN y pasamos los datos a tipo numerico (los que pueden)


df['od'] = df['od'].replace("no se midió", pd.NA)
df['od'] = pd.to_numeric(df['od'], errors='ignore')

df['ph'] = df['ph'].replace("no se midió", pd.NA)
df['ph'] = pd.to_numeric(df['ph'], errors='ignore')

df['colif_fecales_ufc_100ml'] = df['colif_fecales_ufc_100ml'].replace("no se midió", pd.NA) 
df['colif_fecales_ufc_100ml'] = pd.to_numeric(df['colif_fecales_ufc_100ml'], errors='ignore')

df['escher_coli_ufc_100ml'] = df['escher_coli_ufc_100ml'].replace("no se midió", pd.NA)
df['escher_coli_ufc_100ml'] = pd.to_numeric(df['escher_coli_ufc_100ml'], errors='ignore')

df['enteroc_ufc_100ml'] = df['enteroc_ufc_100ml'].replace("no se midió", pd.NA)
df['enteroc_ufc_100ml'] = pd.to_numeric(df['enteroc_ufc_100ml'], errors='ignore')

df['nitrato_mg_l'] = df['nitrato_mg_l'].replace("no se midió", pd.NA)
df['nitrato_mg_l'] = pd.to_numeric(df['nitrato_mg_l'], errors='ignore')

df['nh4_mg_l'] = df['nh4_mg_l'].replace("no se midió", pd.NA)
df['nh4_mg_l'] = pd.to_numeric(df['nh4_mg_l'], errors='ignore')

df['p_total_l_mg_l'] = df['p_total_l_mg_l'].replace("no se midió", pd.NA)
df['p_total_l_mg_l'] = pd.to_numeric(df['p_total_l_mg_l'], errors='ignore')

df['fosf_ortofos_mg_l'] = df['fosf_ortofos_mg_l'].replace("no se midió", pd.NA)
df['fosf_ortofos_mg_l'] = pd.to_numeric(df['fosf_ortofos_mg_l'], errors='ignore')

df['dbo_mg_l'] = df['dbo_mg_l'].replace("no se midió", pd.NA)
df['dbo_mg_l'] = pd.to_numeric(df['dbo_mg_l'], errors='ignore')

df['dqo_mg_l'] = df['dqo_mg_l'].replace("no se midió", pd.NA)
df['dqo_mg_l'] = pd.to_numeric(df['dqo_mg_l'], errors='ignore')

df['turbiedad_ntu'] = df['turbiedad_ntu'].replace("no se midió", pd.NA)
df['turbiedad_ntu'] = pd.to_numeric(df['turbiedad_ntu'], errors='ignore')

df['hidr_deriv_petr_ug_l'] = df['hidr_deriv_petr_ug_l'].replace("no se midió", pd.NA)
df['hidr_deriv_petr_ug_l'] = pd.to_numeric(df['hidr_deriv_petr_ug_l'], errors='ignore')

df['cr_total_mg_l'] = df['cr_total_mg_l'].replace("no se midió", pd.NA)
df['cr_total_mg_l'] = pd.to_numeric(df['cr_total_mg_l'], errors='ignore')

df['cd_total_mg_l'] = df['cd_total_mg_l'].replace("no se midió", pd.NA)
df['cd_total_mg_l'] = pd.to_numeric(df['cd_total_mg_l'], errors='ignore')

df['clorofila_a_ug_l'] = df['clorofila_a_ug_l'].replace("no se midió", pd.NA)
df['clorofila_a_ug_l'] = pd.to_numeric(df['clorofila_a_ug_l'], errors='ignore')

df['microcistina_ug_l'] = df['microcistina_ug_l'].replace("no se midió", pd.NA)
df['microcistina_ug_l'] = pd.to_numeric(df['microcistina_ug_l'], errors='ignore')

df['ica'] = df['ica'].replace("no se midió", pd.NA)
df['ica'] = pd.to_numeric(df['ica'], errors='ignore')


C:\Users\berni\AppData\Local\Temp\ipykernel_6104\3807158427.py:5: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead
  df['od'] = pd.to_numeric(df['od'], errors='ignore')
C:\Users\berni\AppData\Local\Temp\ipykernel_6104\3807158427.py:8: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead
  df['ph'] = pd.to_numeric(df['ph'], errors='ignore')
C:\Users\berni\AppData\Local\Temp\ipykernel_6104\3807158427.py:11: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead
  df['colif_fecales_ufc_100ml'] = pd.to_numeric(df['colif_fecales_ufc_100ml'], errors='ignore')
C:\Users\berni\AppData\Local\Temp\ipykernel_6104\3807158427.py:14: FutureWarning: errors='ignore' is deprecated an

In [96]:
# Corroboramos como van quedando los datos
#df.info()

In [97]:
#Los pasamos a Booleano los que quedaban con (Presencia, Ausencia y "no se midio"), de la misma manera que hicimos con la columna Espuma

# Aca identificamos las la dif entre como estan escritos
#for idx in df['olores'].value_counts().items():
#    print(idx)

In [98]:
# Remplazo los valores q estan escritos distintos para q esten iguales
df['olores'] = df['olores'].replace("Ausencia ", 'Ausencia')
df['olores'] = df['olores'].replace("no se midió", pd.NA)

#Los pasamos a Booleano para mejor manupilacion 
df['olores'] = df['olores'].map({'Presencia': True, 'Ausencia': False})

In [99]:
# Reemplazamos lo "no se midio" por NaN
df['calidad_de_agua'] = df['calidad_de_agua'].replace("no se midió", pd.NA)
#df['calidad_de_agua'].value_counts()

In [100]:
# Aca identificamos las la dif entre como estan escritos
#for idx in df['color'].value_counts().items():
#    print(idx)

In [101]:
# Remplazo los valores q estan escritos distintos para q esten iguales
df['color'] = df['color'].replace("Ausencia ", 'Ausencia')
df['color'] = df['color'].replace("ausenca", 'Ausencia')
df['color'] = df['color'].replace("no se midió", pd.NA)
df['color'] = df['color'].map({'Presencia': True, 'Ausencia': False})


In [102]:
# Aca identificamos las la dif entre como estan escritos
#for idx in df['mat_susp'].value_counts().items():
#    print(idx)

In [103]:
# Remplazo los valores q estan escritos distintos para q esten iguales
df['mat_susp'] = df['mat_susp'].replace("Ausencia ", 'Ausencia')
df['mat_susp'] = df['mat_susp'].replace("no se midió", pd.NA)

#Los pasamos a Booleano para mejor manupilacion 
df['mat_susp'] = df['mat_susp'].map({'Presencia': True, 'Ausencia': False})

In [104]:
# Miramos como quedaron los datos en el dataframe

# df.value_counts(dropna=False)

In [106]:
# Meto en "filas_parcialmente_nan" todas filas que estan en Nulo (excepto las 6 primeras columnas)
# df.isna() me dice que valores son nulos
# sum(axis = 1) suma los nulos pero suma de columna a columna (osea cuantos nulos tiene una fila)
# len(df.columns) - 6  simplemente te dice la longitud (de las columnas) y le resto 6 porq no quiero que tome en cuenta los 6 primeros

filas_parcialmente_nan = df[df.isna().sum(axis=1) == (len(df.columns) - 6)]

In [132]:
# Mostrar las filas con todo NaN
#filas_parcialmente_nan

orden                                     sitios codigo      fecha   año  \
37      10                                   Del Arca  SF015 2021-05-19  2021   
40      11                          Espigón La Farola  SI021 2021-02-24  2021   
41      11                          Espigón La Farola  SI021 2021-05-19  2021   
42      11                          Espigón La Farola  SI021 2021-08-25  2021   
43      11                          Espigón La Farola  SI021 2021-10-27  2021   
69      18                        Parque de los Niños  CA041 2021-05-19  2021   
73      19          Costanera Norte - Espigón Abanico  CA044 2021-05-19  2021   
77      20                         Club de Pescadores  CA046 2021-05-19  2021   
81      21  Reserva Ecológica Costanera Sur - Playita  CA047 2021-05-19  2021   
131     33                          Camping Eva Perón  EN081 2021-08-25  2021   
145     37                      Balneario Palo Blanco  BS092 2021-05-19  2021   
150     38              Diagonal 66 (descarga cloaca)  BS095 2021-05-19  2021   
154     39                         Playa La Bagliardi  BS091 2021-05-19  2021   
157     40                        Balneario Municipal  BS094 2021-05-19  2021   
162     41                          Playa La Balandra  BS093 2021-05-19  2021   

       campaña  tem_agua  tem_aire  od  ph olores color espumas mat_susp  \
37       otoño       NaN       NaN NaN NaN    NaN   NaN     NaN      NaN   
40      verano       NaN       NaN NaN NaN    NaN   NaN     NaN      NaN   
41       otoño       NaN       NaN NaN NaN    NaN   NaN     NaN      NaN   
42    invierno       NaN       NaN NaN NaN    NaN   NaN     NaN      NaN   
43   primavera       NaN       NaN NaN NaN    NaN   NaN     NaN      NaN   
69       otoño       NaN       NaN NaN NaN    NaN   NaN     NaN      NaN   
73       otoño       NaN       NaN NaN NaN    NaN   NaN     NaN      NaN   
77       otoño       NaN       NaN NaN NaN    NaN   NaN     NaN      NaN   
81       otoño       NaN       NaN NaN NaN    NaN   NaN     NaN      NaN   
131   invierno       NaN       NaN NaN NaN    NaN   NaN     NaN      NaN   
145      otoño       NaN       NaN NaN NaN    NaN   NaN     NaN      NaN   
150      otoño       NaN       NaN NaN NaN    NaN   NaN     NaN      NaN   
154      otoño       NaN       NaN NaN NaN    NaN   NaN     NaN      NaN   
157      otoño       NaN       NaN NaN NaN    NaN   NaN     NaN      NaN   
162      otoño       NaN       NaN NaN NaN    NaN   NaN     NaN      NaN   

    colif_fecales_ufc_100ml  escher_coli_ufc_100ml  enteroc_ufc_100ml  \
37                     <NA>                    NaN                NaN   
40                     <NA>                    NaN                NaN   
41                     <NA>                    NaN                NaN   
42                     <NA>                    NaN                NaN   
43                     <NA>                    NaN                NaN   
69                     <NA>                    NaN                NaN   
73                     <NA>                    NaN                NaN   
77                     <NA>                    NaN                NaN   
81                     <NA>                    NaN                NaN   
131                    <NA>                    NaN                NaN   
145                    <NA>                    NaN                NaN   
150                    <NA>                    NaN                NaN   
154                    <NA>                    NaN                NaN   
157                    <NA>                    NaN                NaN   
162                    <NA>                    NaN                NaN   

    nitrato_mg_l nh4_mg_l p_total_l_mg_l fosf_ortofos_mg_l dbo_mg_l dqo_mg_l  \
37          <NA>     <NA>           <NA>              <NA>     <NA>     <NA>   
40          <NA>     <NA>           <NA>              <NA>     <NA>     <NA>   
41          <NA>     <NA>           <NA>              <NA>     <NA>     <NA>   
42

In [108]:
# Eliminar estas filas
df = df.drop(filas_parcialmente_nan.index)

In [109]:
# Eliminamos las columnas de los datos inncesario

df = df.drop('año', axis=1)
df = df.drop('codigo', axis=1)
df = df.drop('orden', axis=1)
df = df.drop('fecha', axis=1)

In [110]:
# Eliminar los símbolos < y > de la columna 'valores'

df['colif_fecales_ufc_100ml'] = df['colif_fecales_ufc_100ml'].str.replace('<', '').str.replace('>', '')
df['nitrato_mg_l'] = df['nitrato_mg_l'].str.replace('<', '').str.replace('>', '')
df['nh4_mg_l'] = df['nh4_mg_l'].str.replace('<', '').str.replace('>', '')
df['p_total_l_mg_l'] = df['p_total_l_mg_l'].str.replace('<', '').str.replace('>', '')
df['fosf_ortofos_mg_l'] = df['fosf_ortofos_mg_l'].str.replace('<', '').str.replace('>', '')
df['dbo_mg_l'] = df['dbo_mg_l'].str.replace('<', '').str.replace('>', '')
df['dqo_mg_l'] = df['dqo_mg_l'].str.replace('<', '').str.replace('>', '')
df['turbiedad_ntu'] = df['turbiedad_ntu'].str.replace('<', '').str.replace('>', '')
df['hidr_deriv_petr_ug_l'] = df['hidr_deriv_petr_ug_l'].str.replace('<', '').str.replace('>', '')
df['cr_total_mg_l'] = df['cr_total_mg_l'].str.replace('<', '').str.replace('>', '')
df['cd_total_mg_l'] = df['cd_total_mg_l'].str.replace('<', '').str.replace('>', '')
df['clorofila_a_ug_l'] = df['clorofila_a_ug_l'].str.replace('<', '').str.replace('>', '')
df['microcistina_ug_l'] = df['microcistina_ug_l'].str.replace('<', '').str.replace('>', '')
df['dqo_mg_l'] = df['dqo_mg_l'].str.replace('<', '').str.replace('>', '')
df['dqo_mg_l'] = df['dqo_mg_l'].str.replace('<', '').str.replace('>', '')

In [111]:
# Y ahora que los valores dejaron de ser string , vuelvo a intentar convertirlos en tipo numerico 

df['od'] = pd.to_numeric(df['od'], errors='ignore')
df['ph'] = pd.to_numeric(df['ph'], errors='ignore')
df['colif_fecales_ufc_100ml'] = pd.to_numeric(df['colif_fecales_ufc_100ml'], errors='ignore')
df['escher_coli_ufc_100ml'] = pd.to_numeric(df['escher_coli_ufc_100ml'], errors='ignore')
df['enteroc_ufc_100ml'] = pd.to_numeric(df['enteroc_ufc_100ml'], errors='ignore')
df['nitrato_mg_l'] = pd.to_numeric(df['nitrato_mg_l'], errors='ignore')
df['nh4_mg_l'] = pd.to_numeric(df['nh4_mg_l'], errors='ignore')
df['p_total_l_mg_l'] = pd.to_numeric(df['p_total_l_mg_l'], errors='ignore')
df['fosf_ortofos_mg_l'] = pd.to_numeric(df['fosf_ortofos_mg_l'], errors='ignore')
df['dbo_mg_l'] = pd.to_numeric(df['dbo_mg_l'], errors='ignore')
df['dqo_mg_l'] = pd.to_numeric(df['dqo_mg_l'], errors='ignore')
df['turbiedad_ntu'] = pd.to_numeric(df['turbiedad_ntu'], errors='ignore')
df['hidr_deriv_petr_ug_l'] = pd.to_numeric(df['hidr_deriv_petr_ug_l'], errors='ignore')
df['cr_total_mg_l'] = pd.to_numeric(df['cr_total_mg_l'], errors='ignore')
df['cd_total_mg_l'] = pd.to_numeric(df['cd_total_mg_l'], errors='ignore')
df['clorofila_a_ug_l'] = pd.to_numeric(df['clorofila_a_ug_l'], errors='ignore')
df['microcistina_ug_l'] = pd.to_numeric(df['microcistina_ug_l'], errors='ignore')
df['ica'] = pd.to_numeric(df['ica'], errors='ignore')

C:\Users\berni\AppData\Local\Temp\ipykernel_6104\3530368319.py:3: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead
  df['od'] = pd.to_numeric(df['od'], errors='ignore')
C:\Users\berni\AppData\Local\Temp\ipykernel_6104\3530368319.py:4: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead
  df['ph'] = pd.to_numeric(df['ph'], errors='ignore')
C:\Users\berni\AppData\Local\Temp\ipykernel_6104\3530368319.py:5: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead
  df['colif_fecales_ufc_100ml'] = pd.to_numeric(df['colif_fecales_ufc_100ml'], errors='ignore')
C:\Users\berni\AppData\Local\Temp\ipykernel_6104\3530368319.py:6: FutureWarning: errors='ignore' is deprecated and 

In [112]:
# Vuelvo a identificar los tipos
#df.info()

In [113]:
# Miramos que van quedando los datos en la tabla
#df.value_counts(dropna=False)

Valores Normales (o Parametro) de la Calidad de Agua:      
    (Esto esta sacado del folleto de calidad de agua, Resolución ACUMAR 46/2017 y la Resolución ADA 42)

* El oxigeno disuelto   -- lo normal es que este en valores superiores a 5 
* El ph -- lo normal es que este entre 6,5 y 9 (aunque podriamos mostrar como zonas delicadas los extremos)
* La temperatura que este por debajo de los 35 
* La Turbiez no puede superar el 100 
* Color, Olor o cosas visibles flotando no tendria que haber
* coliformes fecales hasta 150
* esche hasta 126
* enterococos hasta 33
* nitratos hasta 125 
* amonio hasta 0,5
* fosforo hasta 0,025
* clorofila 50
* microcistina 10
* dbo 10
* hidrocarburos <50
* cromo 50
* cadmio 5

In [115]:
#Buscamos filas con al menos un NaN
filas_parcialmente_nan = df[df.isna().sum(axis=1) >= 1]

In [128]:
# Miramos los datos que nos arrojo

#filas_parcialmente_nan

(162, 31)

In [117]:
# Ordeno los sitios alfabeticamente 
#filas_parcialmente_nan = filas_parcialmente_nan.sort_values(by='sitios', ascending=False)

In [119]:
# Filtrar por un sitio y estación específicos, por ejemplo:
sitio = "Canal Aliviador y Río Lujan"
estacion = "verano"

# Filtrar el DataFrame
filtro = df_copy[(df_copy["sitios"] == sitio) & (df_copy["campaña"] == estacion)]

filtro


orden                       sitios codigo      fecha   año campaña  \
8      3  Canal Aliviador y Río Lujan  TI002 2021-02-24  2021  verano   

   tem_agua  tem_aire    od    ph olores color espumas mat_susp  \
8      24.6      24.0  1.21  7.12    NaN  True    True     True   

  colif_fecales_ufc_100ml  escher_coli_ufc_100ml  enteroc_ufc_100ml  \
8                   20000                18000.0             4600.0   

  nitrato_mg_l nh4_mg_l p_total_l_mg_l fosf_ortofos_mg_l dbo_mg_l dqo_mg_l  \
8          2.8        6           0.68              0.56      6.3      <30   

  turbiedad_ntu hidr_deriv_petr_ug_l cr_total_mg_l cd_total_mg_l  \
8            17                 <100          <5.0        <0.001   

  clorofila_a_ug_l microcistina_ug_l   ica             calidad_de_agua  
8            0.013             <0.15  26.0  Extremadamente deteriorada

In [ ]:
#df = df.sort_values(by='sitios', ascending=False)
#df

 Con respecto a los datos duplicados 2 decisiones posibles, dado a que habla de calidad de agua (personalmente prefiero) decantar por el peor valor, o hacer un promedio entre esos 2 valores
 ademas se podria percibir una disminucion de invierno a primavera (tomando el valor mas alto), es especulativo pero puede ser que justo vieron que se arrojo algun desecho

 PANDEMIA -- Probablemente no 
 para rellenar datos - buscar lugares cercanos geograficamente -- usar valores de las otras estaciones